### PR TITLE
Add GraphQL example

### PR DESCRIPTION
**Pull request by msutkowski**

### CODE REVIEW

File: examples/query/react/graphql/src/mocks/db.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        data: {
+          posts
+        } as { posts: Post[] },
+        per_page,
+        page,
+        total_pages: Math.ceil(db.post.count() / per_page),
+        total: db.post.count(),
+      })
+    )
+  }),
```
Comment (msutkowski) [inline examples/query/react/graphql/src/mocks/db.ts:69]: This seems strange to me, but unfortunately, even though `@mswjs/data` supports pagination on the autogenerated handlers, it doesn't return any pagination information so we have to do this type of thing.

**Review by msutkowski — COMMENTED** [review]
@phryneas If you get time to patch this so it's a more realistic implementation, that'd be great. This seems close, but wonky :sweat:

**Comment by phryneas** [comment]
We *could* ass graphql codegen to generate the types here and provide a more "realistic" example... lemme try

File: examples/query/react/graphql/src/app/services/posts.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+export const api = createApi({
+  baseQuery: graphqlBaseQuery({
+    baseUrl: "/graphql"
+  }),
+  endpoints: (builder) => ({
+    getPosts: builder.query<GetPostsResponse, number | void>({
+      query: (page) => ({
+        body: gql`
+          query GetPosts($page: Int, $per_page: Int) {
+            posts(page: ${page}, per_page: 10) {
```
Comment (phryneas) [inline examples/query/react/graphql/src/app/services/posts.ts:49]:
```suggestion
            posts(page: $page, per_page: 10) {

```

**Pull request merged**